pkg/controller/factory/builder.go

```diff
@@ -336,36 +338,68 @@ func GetRestorePassword(cluster *appsv1alpha1.Cluster, synthesizedComp *componen
 }
 
 // GetRestoreSystemAccountPassword gets restore password if exists during recovery.
-func GetRestoreSystemAccountPassword(annotations map[string]string, componentName, accountName string) string {
+func GetRestoreSystemAccountPassword(
+	ctx context.Context,
+	cli client.Reader,
+	annotations map[string]string,
+	componentName,
+	accountName string,
+) ([]byte, error) {
 	valueString := annotations[constant.RestoreFromBackupAnnotationKey]
 	if len(valueString) == 0 {
-		return ""
+		return nil, nil
 	}
 	backupMap := map[string]map[string]string{}
 	err := json.Unmarshal([]byte(valueString), &backupMap)
 	if err != nil {
-		return ""
+		return nil, err
 	}
 	backupSource, ok := backupMap[componentName]
 	if !ok {
-		return ""
+		// try to find the first available backup source, in case backup is done
```

Comment:
This logic is not in the original PR, we should see if there's a better way, or add it to 1.0 too.